datastore/service.go

```diff
@@ -37,19 +39,30 @@ func (srv Service) GetAll(ctx context.Context, namespace models.NamespaceSpec, d
 func (srv Service) CreateResource(ctx context.Context, namespace models.NamespaceSpec, resourceSpecs []models.ResourceSpec, obs progress.Observer) error {
 	runner := parallel.NewRunner(parallel.WithLimit(ConcurrentLimit), parallel.WithTicket(ConcurrentTicketPerSec))
 	for _, resourceSpec := range resourceSpecs {
-		currentSpec := resourceSpec
-		repo := srv.resourceRepoFactory.New(namespace, currentSpec.Datastore)
+		incomingSpec := resourceSpec
+		repo := srv.resourceRepoFactory.New(namespace, incomingSpec.Datastore)
 		runner.Add(func() (interface{}, error) {
-			if err := repo.Save(ctx, currentSpec); err != nil {
+			proceed, err := srv.isProceedToSave(ctx, repo, incomingSpec)
```

Comment:
cmiiw, I can observe that incoming spec will not have id and urn fields set so it will always save. Why not go with an equals comparision instead of hash instead of have has computation logic lets just go with equals comparision